### PR TITLE
Add MongoCollection Uniqueness compatibility

### PR DESCRIPTION
**Pull request by Jurigag**
_Hello!_

* Type: bug fix
* Link to issue: https://github.com/phalcon/incubator/issues/700

This pull request affects the following components: **(please check boxes)**

* [x] Library
* Code Style
* Documentation
* [x] Testing

**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/phalcon/incubator/blob/master/CONTRIBUTING.md)?
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I wrote some tests for this PR.

Small description of change: this is adding compability for MongoCollection to work with `Phalcon\Validation\Validator\Uniqueness` since from phalcon 3.1.x Collection can use `Phalcon\Validation` component.

Thanks

### CODE REVIEW

File: tests/_data/collections/Robots.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: User
+ * Date: 2017-03-09
+ * Time: 17:11
+ */
```
Comment (sergeyklay) [inline tests/_data/collections/Robots.php]: Could you please remove this

File: tests/unit/Mvc/Collection/ValidationCest.php [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+use UnitTester;
+
+/**
+ * \Phalcon\Test\Mvc\Collection\ValidationCest
+ * Tests for Phalcon\Mvc\MongoCollection component
+ *
+ * @copyright (c) 2011-2017 Phalcon Team
+ * @link      https://www.phalconphp.com
+ * @author    Wojciech Ślawski <jurigag@gmail.com>
+ * @package   Phalcon\Test\Mvc
```
Comment (sergeyklay) [inline tests/unit/Mvc/Collection/ValidationCest.php]: `Phalcon\Test\Mvc\Collection`

**Comment by Jurigag** [comment]
I have no idea why tests are failing.

File: tests/_data/collections/Users.php [hunk 3]
```diff
@@ -0,0 +1,39 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: User
+ * Date: 2017-03-09
+ * Time: 17:07
+ */
```
Comment (sergeyklay) [inline tests/_data/collections/Users.php]: Could you please remove this

File: tests/unit/Mvc/Collection/Helpers/Validation.php [hunk 4]
```diff
@@ -0,0 +1,216 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: User
+ * Date: 2017-03-09
+ * Time: 17:06
+ */
```
Comment (sergeyklay) [inline tests/unit/Mvc/Collection/Helpers/Validation.php]: Could you please remove this

**Pull request merged**